tests/integration/test_charm.py

```diff
@@ -14,94 +14,90 @@
         }
     )
-    await ops_test.model.wait_for_idle(status="active")
+    await model.wait_for_idle(status="active", raise_on_error=False)
+
+    unit_informations = json.loads(
+        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
+    )
+    mattermost_ip = unit_informations[app.units[0].name]["address"]
 
     # create a user
-    cmd = "MMCTL_LOCAL_SOCKET_PATH=/mattermost/run/local.socket /mattermost/bin/mmctl --local user create --email test@test.test --username test --password thisisabadpassword"
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
+    cmd = f"MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl --local user create --email {test_user['login_id']} --username test --password {test_user['password']}"
+    await ops_test.juju("run", "--application", app.name, cmd)
 
     # login to the API
-    data = {"login_id": "test@test.test", "password": "thisisabadpassword"}
-    cmd = f"curl -sid '{json.dumps(data)}' http://localhost:8065/api/v4/users/login"
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
-    token = ""
-    for line in output[1].splitlines():
-        if m := re.match(r"Token: (\w+)", line):
-            token = m.group(1)
-    print(token)
+    response = requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/users/login", data=json.dumps(test_user)
+    )
+    headers = {"authorization": f"Bearer {response.headers['Token']}"}
 
     # create a team
     data = {"name": "test", "display_name": "test", "type": "O"}
-    cmd = f"curl -XPOST -sd '{json.dumps(data)}' -H 'authorization: Bearer {token}' http://localhost:8065/api/v4/teams"
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
-    team = json.loads(output[1])
+    response = requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/teams", data=json.dumps(data), headers=headers
+    )
+    team = response.json()
 
     # create a channel
     data = {"team_id": team["id"], "name": "test", "display_name": "test", "type": "O"}
-    cmd = f"curl -XPOST -sd '{json.dumps(data)}' -H 'authorization: Bearer {token}' http://localhost:8065/api/v4/channels"
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
-    channel = json.loads(output[1])
-
-    # upload a file
-    cmd = (
-        "curl -F 'files=@/etc/os-release' -F 'channel_id="
-        + channel["id"]
-        + "' -H 'authorization: Bearer "
-        + token
-        + "' http://localhost:8065/api/v4/files"
+    response = requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/channels", data=json.dumps(data), headers=headers
     )
-    output = await ops_test.juju("run", "--application", app.name, cmd)
-    print(output)
+    channel = response.json()
+
+    # create a test file
+    test_file = tmp_path / "test_file.txt"
```

Comment:
The string "test_file.txt" is used multiple times, I'd make it a variable

Reply:
variable added